app/uk/gov/hmrc/cataloguefrontend/model/Environment.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cataloguefrontend.model
+
+import play.api.mvc.{PathBindable, QueryStringBindable}
+
+sealed trait Environment { def asString: String }
+
+object Environment {
+  case object Production      extends Environment { val asString = "production"   }
+  case object ExternalTest    extends Environment { val asString = "externaltest" }
+  case object Staging         extends Environment { val asString = "staging"      }
+  case object QA              extends Environment { val asString = "qa"           }
+  case object Integration     extends Environment { val asString = "integration"  }
+  case object Development     extends Environment { val asString = "development"  }
+
+  val values: List[Environment] = List(Production, ExternalTest, Staging, QA, Integration, Development)
+
+  def parse(s: String): Option[Environment] =
+    values.find(_.asString == s)
```

Comment:
Is it worth doing a `.toLowerCase` here?

Reply:
TeamsAndRepositories also has spaces e.g. `"External Test"`. I think being restrictive means we actually know what the inputs from different boundaries look like, rather than going through a load of transformations.